generate.js

```diff
@@ -1,76 +1,54 @@
 'use strict';
 var fs = require('fs');
 var path = require('path');
-var zlib = require('zlib');
-var tar = require('tar');
 var got = require('got');
-var cv2json = require('convert-json');
-var globby = require('globby');
-var del = require('del');
 
 // config
-var url = 'http://git.spdx.org/?p=license-list.git;a=snapshot;h=HEAD;sf=tgz';
-var folder = 'licenses';
-var ignored = [
-	'README.txt',
-	'Updating the SPDX Licenses.txt'
-];
-var colValue = 'v';
+var url = 'https://spdx.org/licenses/licenses.json';
 var licensesJson = {};
 
-// process the excel sheet and generate json files from it
-function processXls() {
-	cv2json.xls(globby.sync('licenses/spdx_licenselist_*.xls')[0], function (err, result) {
-		if (err) {
-			throw err;
-		}
+function getFullDetails(array, index) {
+	if (typeof(index) === 'undefined') {
+		index = 0;
+	}
 
-		var licenses = result.Sheets.licenses;
-		for (var key in licenses) {
-			var row = key.match(/^A(\d+)$/);
-			if (row && row[1] > 1) {
-				var name = licenses[row[0]][colValue].trim();
-				var identifier = licenses['B' + row[1]][colValue].trim();
-				var url = licenses['C' + row[1]] ? licenses['C' + row[1]][colValue] : null;
-				var approved = licenses['E' + row[1]] ? licenses['E' + row[1]][colValue] === 'YES' : false;
+	var license = array[index];
```

Comment:
Are you familiar with promises or not all? If yes, continue reading, if not read [this](https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html) first.

Promise.all() can be used to bundle an array of promises and wait untill all the promises in the array complete.

```js
function download(index) {
    return got(`url/${index}`)
        .then(result => {
            // handle result
        });
}

const array = [0, 1, 2, 3];

Promise.all(array.map(download)).then(result => {
    // result is an array of all the got requests
});
```

That's very basic how you can leverage `Promise.all` here.